src/zocalo/service/mailer.py

```diff
@@ -0,0 +1,153 @@
+from __future__ import annotations
+
+import email.message
+import pprint
+import smtplib
+
+import workflows.recipe
+from workflows.services.common_service import CommonService
+
+import zocalo.configuration
+
+
+class _SafeDict(dict):
+    """A dictionary that returns undefined keys as {keyname}.
+    This can be used to selectively replace variables in datastructures."""
+
+    def __missing__(self, key):
+        return "{" + key + "}"
+
+
+class Mailer(CommonService):
+    """A service that generates emails from messages."""
+
+    # Human readable service name
+    _service_name = "Mail Notifications"
+
+    # Logger name
+    _logger_name = "zocalo.services.mailer"
+
+    def initializing(self):
+        """Subscribe to the Mail notification queue.
+        Received messages must be acknowledged."""
+        self.log.debug("Mail notifications starting")
+
+        zc = zocalo.configuration.from_file()
+        zc.activate()
+        if not zc.smtp:
+            raise zocalo.ConfigurationError(
+                "There are no SMTP settings configured in your environment"
+            )
+        self._smtp = zc.smtp
```

Comment:
```suggestion
        if not self._environment.get("config"):
            raise zocalo.ConfigurationError(
                "No Zocalo configuration loaded"
            )
        if not self._environment["config"].smtp:
            raise zocalo.ConfigurationError(
                "There are no SMTP settings configured in your environment"
            )
        self._smtp = self._environment["config"].smtp
```

Use the loaded configuration rather than attempting to load another instance.

Reply:
Thanks, I've updated this based on your suggestions, using the convenience `self.config` property that handles the `self._environment.get("config") bit.
https://github.com/DiamondLightSource/python-workflows/blob/7da5391fd38c1183b10be43bac32dfc39ccf9f64/src/workflows/services/common_service.py#L162-L164